### ISSUE TITLE
High memory usage for very big directories

### ISSUE THREAD

**Issue by mrmlnc** [mrmlnc/fast-glob#204]
### Environment

  * OS Version: Windows 10 (1903)
  * Node.js Version: 12.4.0

### Actual behavior

```
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
```

### Expected behavior

I see entries.

### Steps to reproduce

1. Find directory with 2-4kk entries.
2. Run `fast-glob` with `**` pattern

**Comment by mrmlnc** [comment]
Initial analysis:

* The `_transform` method call `replace` method for each entry. Memory profiling: 786MB
* The `setBasePathForEntryPath` method call `startsWith` method for each entry. Memory profiling: 180MB

**Comment by mrmlnc** [comment]
Will be fixed with `fast-glob@3.0.2`.
